### PR TITLE
Run one process per linter (again)

### LINKED ISSUE #312: Partitioning can result in 0 paths, but still executes
eg. With an invalid path:

```
DEBUG: executing /Users/aat/.go/bin/gotype -e
```
Note the lack of any paths, which causes the linter to fail.

cc @dnephin

### LINKED ISSUE #313: goconst only accepts one directory at a time
```
  goconst ARGS <directory>
```

cc @dnephin

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #312, #313 and #307

### CODE REVIEW

**Comment by alecthomas** [comment]
Hmm annoying. I assumed that when I merged #314 this would not have any conflicts ... :(

**Comment by dnephin** [comment]
I was able to rebase this without any conflicts. Could it maybe be that you have github branch checks enabled? I think I've seen cases where "conflict" is reported by github, but it's really a github branch check that notices that the PR doesn't have the master commit included.

Either way, this is rebased :)

File: main.go [hunk 1]
```diff
@@ -319,6 +317,29 @@ func expandPaths(paths, skip []string) []string {
...
+	for _, name := range skip {
+		filter[name] = true
+	}
+
+	return func(path string) bool {
+		base := filepath.Base(path)
+		if filter[base] || filter[path] {
+			return true
+		}
+		return base != "." && base != ".." && strings.ContainsAny(base[0:1], "_.")
```
Comment (dnephin) [inline main.go]: I don't believe so. The old code had this same logic.

I've added a test case so you can confirm.

Comment (alecthomas) [inline main.go]: Ah sorry, you're right.

**Comment by alecthomas** [comment]
Excellent, thanks!

**Comment by teknico** [comment]
This did not fix #307 unfortunately.

**Pull request merged**